### DISCUSSION TITLE
Meshlab Export of UV

### DISCUSSION THREAD

**Discussion by poro56** [cnr-isti-vclab/meshlab#1260] [Q&A]
When I exported my Model from Meshlab, I lost my UV Maps even without applying a Filter or any other action.

I tried ply, obj and dae.

The same problem happens in pymeshlab.

What can I do to get my UV Map exported even when I did a non-mesh-destructive operation?
Perhaps for pymeshlab too.

**Comment by alemuntoni** [comment]
Did you check the per vertex or per wedge `TexCoord` checkbox in the save dialog?

**Comment by poro56** [comment]
Yes, I did. The same in pymeshlab. 

I tried another Mesh because the files before were exported from Blender when I exported it as dae. Meshlab Crashes.

https://user-images.githubusercontent.com/106281427/170492948-ab88a288-9f30-415e-a911-11d12fe083b0.mp4

**Reply by alemuntoni** [reply]
could you please attach a file in order to replicate the issue?

**Reply by poro56** [reply]
Yes. Give me a minute.

**Reply by poro56** [reply]
Obj Export works with this file.

**Comment by poro56** [comment]
Perhaps this question will solve my problem.

When I import a Mesh with Quads or N-Gons to Meshlab, does Meshlab destroy the UVs by its Quad/N-Gons to Tris Conversion?

At first, I did not consider the option because when I read the Feature/Support/Reference page on meshlab.net and the pymeshlab documentation on readthedocs.io, it was not mentioned that UVs would be destroyed by a Quad to Tris. Still, all the Meshes I used were Quad Meshes.

Perhaps this is the issue?

**Reply by poro56** [reply]
No. This was not the problem. I triangulated all the faces and exported and still have the UV issues. I tried my assumption.

Can there be a compatibility issue between Blender Exports and Meshlabs? (For example, I know  Blender had a problem with per Wedge UV Coordinates by PLY Files.

In pymeshlab I tried through convert_perwedge_uv_into_pervertex_uv to archive this but currently, no success.

**Reply by poro56** [reply]
I found a solution in pymeshlab.

**Reply by emadpres** [reply]
@poro56 Can you please share the solution with others?
